src/ape_ethereum/transactions.py

```diff
@@ -207,6 +208,26 @@ def failed(self) -> bool:
     def call_tree(self) -> Optional[CallTreeNode]:
         return self.provider.get_call_tree(self.txn_hash)
 
+    @cached_property
+    def debug_logs_typed(self) -> List[Tuple[Any]]:
+        """
+        Extract messages to console outputted by contracts via print() or console.log() statements
+        """
+
+        try:
+            assert self.call_tree is not None
```

Comment:
instead of assert + catching assertion error, check if it None later on.

When compiling python to an optimized mode, asserts are removed, they should not be used this way.

Reply:
Spaced on this.  Don't know why I like this pattern so much.  Will change.